lbrynet/extras/daemon/Daemon.py

```diff
@@ -3328,173 +3328,122 @@ def jsonrpc_routing_table_get(self):
+            comment_create  (<comment> | --comment=<comment>)
+                            (<claim_id> | --claim_id=<claim_id>)
+                            [--channel_id=<channel_id>]
+                            [--channel_name=<channel_name>]
+                            [--parent_id=<parent_id>]
 
         Options:
-            --parent_comment_id=<parent_comment_id>  : (int) The ID of a comment to make a response to
+            --comment=<comment>         : (str) Comment to be made, should be at most 2000 characters.
+            --claim_id=<claim_id>       : (str) The ID of the claim on which the comment should be made on
+            --parent_id=<parent_id>     : (str) The ID of a comment to make a response to
+            --channel_id=<channel_id>   : (str) The ID of the channel you want to post under
+            --channel_name=<channel_name>   : (str) The channel you want to post as, prepend with a '@'
+
 
         Returns:
-            (dict) Comment object if successfully made
-        """
-        if not 1 < len(message) <= 2000:
-            raise Exception(f'Message length ({len(message)}) needs to be between 2 and 2000 chars')
-        url = self.conf.comment_server
-        if parent_comment_id is not None:
-            comment_id = await jsonrpc_post(url, 'reply', parent_id=parent_comment_id,
-                                            poster=channel_id, message=message)
-        else:
-            claim_data = await self.jsonrpc_claim_search(claim_id=claim_id)
-            if 'error' not in claim_data and claim_data['total_pages'] == 1:
-                uri = claim_data['items'][0].permanent_url
-                comment_id = await jsonrpc_post(url, 'comment', uri=uri,
-                                                poster=channel_id, message=message)
-            else:
-                raise Exception(f"permanent_url is not in the claim_data {claim_data}\n"
-                                f"The given claim_id ({claim_id}) may be invalid")
-        return await jsonrpc_post(url, 'get_comment_data', comm_index=comment_id, better_keys=True)
+            (dict) Comment object if successfully made, (None) otherwise
+            {
+                "comment":      (str) The actual string as inputted by the user,
+                "comment_id":   (str) The Comment's unique identifier,
+                "channel_name": (str) Name of the channel this was posted under, prepended with a '@',
+                "channel_id":   (str) The Channel Claim ID that this comeent was posted under,
+                "signature":    (str) The signature of the comment,
+                "channel_url":  (str) Channel's URI in the ClaimTrie,
+                "parent_id":    (str) Comment this is replying to, (None) if this is the root,
+                "timestamp":    (int) The time at which comment was entered into the server at, in nanoseconds.
+            }
+        """
+        if bool(channel_name) ^ bool(channel_id):
```

Comment:
dafuqdis? Is this being submitted to a Python Obfuscated Code competition? It's not even consistent with an `if channel_id and channel_name` later in the same method.

This should just be:
```
if not channel or not channel_id:
```

Reply:
It's `channel_id XOR channel_name`. If only one is provided then we have to search for the other one. If both are provided, we don't have to look; if neither are provided, we don't have to look.